### PR TITLE
Use max_items as bound in tower-batch

### LINKED ISSUE #1670: Use the correct bound for the semaphore in tower-batch
**Is your feature request related to a problem? Please describe.**

`tower-batch` uses a semaphore bound of 1, but `tower::Buffer` uses the supplied bound:
https://github.com/ZcashFoundation/zebra/blob/main/tower-batch/src/service.rs#L60
https://docs.rs/tower/0.4.4/src/tower/buffer/service.rs.html#86

**Describe the solution you'd like**

We should use the supplied bound for the semaphore, so `tower-batch` can handle the expected number of concurrent callers.

**Describe alternatives you've considered**

Do nothing: the code might hang or otherwise be slow.

**Additional context**

This issue was discovered during the review in #1593. It could be a cause of the hangs in #1435, so it's a high priority.

### PR DESCRIPTION
**Pull request by oxarbitrage**
## Motivation

We want to change the max number of requests of the semaphore in zebra-batch.

Closes https://github.com/ZcashFoundation/zebra/issues/1670

## Solution

I discussed with @teor2345  that the best by now for this will be to use `max_items` argument as `bound`.

## Review

@teor2345

### CODE REVIEW

**Comment by oxarbitrage** [comment]
<strike>
I am wondering if we need:

https://github.com/tower-rs/tower/blob/master/tower/src/semaphore.rs#L93-L100

If so, we might have to use an older version:

https://github.com/tower-rs/tower/commit/43c44922af2254fed768832aae9ef8cf4f33410b#diff-962111ba9c46d1673d59e205b3c1a9ff6996e90e23863e91a192d148e29512bc
</strike>

We changed the approach on how to do this, this comment do not apply anymore.

File: tower-batch/src/service.rs [hunk 1]
```diff
@@ -57,8 +57,8 @@ where
         T::Error: Send + Sync,
         Request: Send + 'static,
     {
-        // XXX(hdevalence): is this bound good
-        let bound = 1;
+        // We use the max_items as maximum number of requests.
```
Comment (oxarbitrage) [inline tower-batch/src/service.rs]: Do we need to add to this comment more info like why do we think is a good idea to have the bound the same as the max_items ?

Comment (teor2345) [inline tower-batch/src/service.rs]:
```suggestion
        // The semaphore bound limits the maximum number of concurrent requests
        // (specifically, requests which got a `Ready` from `poll_ready`, but haven't
        // used their semaphore reservation in a `call` yet).
        // We choose a bound that allows callers to check readiness for every item in
        // a batch, then actually submit those items.
```

**Review by teor2345 — APPROVED** [review]
We're good here

**Pull request merged**